src/components/TodoApp.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { useTodo } from './TodosContext';
+
+export const TodoApp: React.FC = () => {
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [showError, setShowError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const {
+    todos,
+    addTodo,
+    toogleAll,
+    deleteComplitedTodo,
+  } = useTodo();
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const handleInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      if (newTodoTitle.trim()) {
+        addTodo(newTodoTitle);
+        setNewTodoTitle('');
+        setShowError(false);
+      } else {
+        setShowError(true);
+        setErrorMessage('Todo title cannot be empty.');
+        setTimeout(() => {
+          setShowError(false);
+        }, 2000);
+      }
+    }
+  };
+
+  const toggleAllCheck = todos.every(todo => todo.completed);
+
+  const todosUncomplited = todos.filter(
+    todo => !todo.completed,
+  ).length;
+
+  const someTodoIsComplited = todos.some(todo => todo.completed);
```

Comment:
make sense to use useMemo for toggleAllCheck, todosUncomplited and someTodoIsComplited because it's essential to performance